src/Workspaces/CSharpTest/Formatting/FormattingTests.cs

```diff
@@ -5219,6 +5219,32 @@ void M()
             await AssertFormatAsync(expectedCode, code, changedOptionSet: changingOptions);
         }
 
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.Formatting)]
+        [WorkItem(34683, "https://github.com/dotnet/roslyn/issues/34683")]
+        public async Task FormatRecursivePattern_InBinaryOperation()
+        {
+            var changingOptions = new Dictionary<OptionKey, object>();
+            changingOptions.Add(CSharpFormattingOptions.SpaceWithinMethodCallParentheses, true);
+            var code = @"class C
+{
+    void M()
+    {
+        return
+            typeWithAnnotations is { } && true;
+    }
+}";
+            var expectedCode = @"class C
```

Comment:
❔ Are there any changes here? I spend a short bit of time looking but could not spot the difference. If there are no changes, consider using:

```csharp
var expectedCode = code;
```

Reply:
No change. I will clarify as you suggested.

Reply:
Sorry about that

Reply:
No need to apologize 😄